connectors/sources/sharepoint_online.py

```diff
@@ -663,125 +719,268 @@ async def validate_config(self):
-                        # Also we need to remember original ID because when a document
-                        # is yielded, its "id" field is overwritten with content of "_id" field
-                        list_item_natural_id = list_item["id"]
-                        list_item["_id"] = f"{site_list['id']}-{list_item['id']}"
-                        list_item["object_type"] = "list_item"
-                        list_item["_tempfile_suffix"] = os.path.splitext(
-                            list_item.get("FileName", "")
-                        )[-1]
-
-                        content_type = list_item["contentType"]["name"]
-
-                        if content_type in [
-                            "Web Template Extensions",
-                            "Client Side Component Manifests",
-                        ]:  # TODO: make it more flexible. For now I ignore them cause they 404 all the time
-                            continue
-
-                        yield list_item, None
-
-                        if "Attachments" in list_item["fields"]:
-                            async for list_item_attachment in self.client.site_list_item_attachments(
-                                site["webUrl"], site_list["name"], list_item_natural_id
-                            ):
-                                list_item_attachment["_id"] = list_item_attachment[
-                                    "odata.id"
-                                ]
-                                list_item_attachment[
-                                    "object_type"
-                                ] = "list_item_attachment"
-                                list_item_attachment["_timestamp"] = list_item[
-                                    "lastModifiedDateTime"
-                                ]
-                                attachment_download_func = partial(
-                                    self.get_attachment_content,
-                                    list_item_attachment,
-                                )
-                                yield list_item_attachment, attachment_download_func
-
-                async for site_page in self.client.site_pages(site["webUrl"]):
-                    site_page["_id"] = site_page[
-                        "odata.id"
-                    ]  # Apparantly site_page["GUID"] is not globally unique
-                    site_page["object_type"] = "site_page"
-
-                    for html_field in ["LayoutWebpartsContent", "CanvasContent1"]:
-                        if html_field in site_page:
-                            site_page[html_field] = html_to_text(site_page[html_field])
+                        yield list_item, download_func
 
+                # Sync site pages
+                async for site_page in self.site_pages(site["webUrl"]):
                     yield site_page, None
 
+    async def get_docs_incrementally(self, sync_cursor, filtering=None):
+        self._sync_cursor = sync_cursor
```

Comment:
I'd suggest assigning `sync_cursor` to a separate variable. `self._sync_cursor` is supposed to be updated to the connector when the sync is done successfully, while the `sync_cursor` is the cursor from which the incremental sync starts.